tracer/src/Datadog.Trace/TracerManagerFactory.cs

```diff
@@ -349,13 +352,80 @@ protected virtual ISpanSampler GetSpanSampler(TracerSettings settings)
         protected virtual IAgentWriter GetAgentWriter(TracerSettings settings, IDogStatsd statsd, Action<Dictionary<string, float>> updateSampleRates, IDiscoveryService discoveryService)
         {
             var apiRequestFactory = TracesTransportStrategy.Get(settings.Exporter);
-            var api = new Api(apiRequestFactory, statsd, updateSampleRates, settings.Exporter.PartialFlushEnabled);
+            var api = GetApi(settings, statsd, updateSampleRates, apiRequestFactory, settings.Exporter.PartialFlushEnabled);
 
             var statsAggregator = StatsAggregator.Create(api, settings, discoveryService);
 
             return new AgentWriter(api, statsAggregator, statsd, maxBufferSize: settings.TraceBufferSize, batchInterval: settings.TraceBatchInterval, apmTracingEnabled: settings.ApmTracingEnabled);
         }
 
+        private IApi GetApi(TracerSettings settings, IDogStatsd statsd, Action<Dictionary<string, float>> updateSampleRates, IApiRequestFactory apiRequestFactory, bool partialFlushEnabled)
+        {
+            if (settings.DataPipelineEnabled)
+            {
+                try
+                {
+                    var telemetrySettings = TelemetrySettings.FromSource(GlobalConfigurationSource.Instance, TelemetryFactory.Config, settings, isAgentAvailable: null);
```

Comment:
nit: for later, we should probably refactor this so we're not recalculating all the telemetry settings _again_ - we already know all the details + this doesn't seem to take into account fine-grained switches like enabling/disabling metrics etc 🤔 

I think it's fine for this PR, we should just try to improve it later